shared/chat/conversation/input-area/normal/mention-input.desktop.js

```diff
@@ -0,0 +1,237 @@
+// @flow
+import * as React from 'react'
+import PlatformInput from './platform-input'
+import {type MentionInputProps} from './types'
+import {Input} from '../../../../common-adapters'
+
+type MentionState = {
+  upArrowCounter: number,
+  downArrowCounter: number,
+  pickSelectedCounter: number,
+  mentionFilter: string,
+  channelMentionFilter: string,
+  mentionPopupOpen: boolean,
+  channelMentionPopupOpen: boolean,
+}
+
+class MentionInput extends React.Component<MentionInputProps, MentionState> {
+  state: MentionState = {
```

Comment:
didn't change most of the logic, but converted from a HOC to a wrapper component, so lotta whitespace diffs